frontend/kesaseteli/youth/src/components/forms/AdditionalInfoForm.tsx

```diff
@@ -0,0 +1,82 @@
+import useCreateAdditionalInfoQuery from 'kesaseteli/youth/hooks/backend/useCreateAdditionalInfoQuery';
+import useRegisterInput from 'kesaseteli/youth/hooks/useRegisterInput';
+import { ADDITIONAL_INFO_REASON_TYPE } from 'kesaseteli-shared/constants/additional-info-reason-type';
+import AdditionalInfoFormData from 'kesaseteli-shared/types/additional-info-form-data';
+import AdditionalInfoReasonOption from 'kesaseteli-shared/types/additional-info-reason-option';
+import CreatedYouthApplication from 'kesaseteli-shared/types/created-youth-application';
+import { useTranslation } from 'next-i18next';
+import React from 'react';
+import SaveFormButton from 'shared/components/forms/buttons/SaveFormButton';
+import Heading from 'shared/components/forms/heading/Heading';
+import MultiSelectDropdown from 'shared/components/forms/inputs/MultiSelectDropdown';
+import TextInput from 'shared/components/forms/inputs/TextInput';
+import FormSection from 'shared/components/forms/section/FormSection';
+import { $GridCell, $Hr } from 'shared/components/forms/section/FormSection.sc';
+import { $Notification } from 'shared/components/notification/Notification.sc';
+import { useTheme } from 'styled-components';
+
+type Props = {
+  applicationId: CreatedYouthApplication['id'];
+};
+
+const AdditionalInfoForm: React.FC<Props> = ({ applicationId }) => {
+  const { t } = useTranslation();
+
+  const register = useRegisterInput<AdditionalInfoFormData>('additionalInfo');
+  const reasons: AdditionalInfoReasonOption[] = React.useMemo(
+    () =>
+      ADDITIONAL_INFO_REASON_TYPE.map((type) => ({
+        name: type,
+        label: t(`common:additionalInfo.reasons.${type}`),
+      })),
+    [t]
+  );
```

Comment:
A discussion: I came across some articles about overusing useMemo, React doc also mentions that useMemo helps to avoid expensive calculations. This is a simple one, so using useMemo might not optimize performance but can use more memory. 1 reference: https://medium.com/swlh/should-you-use-usememo-in-react-a-benchmarked-analysis-159faf6609b7, but there are more you can find

Reply:
You're probably right. I've read about the useMemo/useCallback that it isn't always good to use them but I still had. In this pr was at least one place that I had to use it to make the code work. Anyway it seems that useMemo is good with extensive lists or other extensive calculation, while with useCallback this [article](https://www.alexeyshmalko.com/20210128041144/) says:

> Two legible use cases for useCallback are:

- the child component uses React.memo or shouldComponentUpdate (→ [React.useCallback breaks encapsulation](https://www.alexeyshmalko.com/20210128042138/))
- the callback is used as a dependency in useEffect

Reply:
I still feel a little lazy to touch the code now to remove these useMemos or callbacks, but it's perhaps something to do in the future when refactoring. There was one place where it was needed or rendering didnt work correctly so removing all now from this pr probably would get that problem back for now

Reply:
yeah just want to have a discussion with you about this, thank you for the insight!